### PR TITLE
Implement a metrics resource using Vumi blinkenlights metrics publishers.

### PR DESCRIPTION
**Pull request by hodgestar**

### CODE REVIEW

File: go/vumitools/app_worker.py [hunk 1]
```diff
@@ -230,6 +230,16 @@ def publish_conversation_metric(self, conversation, name, value, agg=None):
             conversation.user_account.key, conversation.key, name)
         self.publish_metric(name, value, agg)
 
+    def publish_account_metric(self, user_account, store, name, value,
+                               agg=None):
+        if isinstance(user_account, basestring):
+            user_account_key = user_account
+        else:
+            user_account_key = user_account.key
```
Comment (jerith) [inline go/vumitools/app_worker.py]: Do we really need to accept both of these here? I'd prefer to have the caller extract the key if necessary.

Comment (hodgestar) [inline go/vumitools/app_worker.py]: I don't really mind -- I was just trying to fit in with `publish_conversation_metric` which takes a `conversation` but not require people to look up the account if they only have the key.

**Comment by jerith** [comment]
:+1:

**Comment by hodgestar** [comment]
This replaced https://github.com/praekelt/vumi-go/pull/121.

File: go/apps/jsbox/tests/test_metrics.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    def test_handle_fire(self):
+        cmd = SandboxCommand(metric="foo", value=1.5, agg='sum')
+        reply = self.resource.handle_fire(self.dummy_api, cmd)
+        self.check_reply(reply, cmd, True)
+        self.check_publish('default', 'foo', 1.5, self.SUM)
+
+    def _test_error(self, cmd, expected_error):
+        with LogCatcher() as lc:
+            reply = self.resource.handle_fire(self.dummy_api, cmd)
+            self.assertEqual(lc.messages(), [expected_error])
```
Comment (smn) [inline go/apps/jsbox/tests/test_metrics.py:126]: Why isn't this in `lc.errors` ?

Comment (hodgestar) [inline go/apps/jsbox/tests/test_metrics.py:126]: Because we use `log.warning` to log the error. :)

**Pull request merged**